src/components/ReportActionItem/MoneyReportView.tsx

```diff
@@ -65,30 +58,37 @@ function MoneyReportView({report, policyReportFields, shouldShowHorizontalRule,
         StyleUtils.getColorStyle(theme.textSupporting),
     ];
 
-    const sortedPolicyReportFields = useMemo(
-        () => policyReportFields.sort(({orderWeight: firstOrderWeight}, {orderWeight: secondOrderWeight}) => firstOrderWeight - secondOrderWeight),
-        [policyReportFields],
-    );
-    const isAdmin = ReportUtils.isPolicyAdmin(report.policyID ?? '', policies);
+    const sortedPolicyReportFields = useMemo<PolicyReportField[]>((): PolicyReportField[] => {
+        const reportFields = Object.values(report.reportFields ?? {});
+        const mergedFieldIds = Array.from(new Set([...policyReportFields.map(({fieldID}) => fieldID), ...reportFields.map(({fieldID}) => fieldID)]));
+        const mergedFields = mergedFieldIds.map((id) => report?.reportFields?.[id] ?? policyReportFields.find(({fieldID}) => fieldID === id)) as PolicyReportField[];
+        const allReportFields = isSettled ? reportFields : mergedFields;
+        return allReportFields.sort(({orderWeight: firstOrderWeight}, {orderWeight: secondOrderWeight}) => firstOrderWeight - secondOrderWeight);
```

Comment:
Could we move this to a utility method, and then add some comments for context on when we use the reportFields vs merged fields?

Reply:
Yep, added!